front/app/resources/GetProjectAllowedInputTopics.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+// hooks
+import {
```

Comment:
Hm, so we are creating this resource component with this PR - can you explain why not use a hook instead?

Reply:
The main reason is that we still have two class components that need this data: the `components/admin/PostManager` and the `components/IdeaForm`. 

For the former, the hardest thing is to rewrite the logic inside of the `Data` wrapper component that uses `react-adopt`, where we render all kinds of different resource components based on the `type` prop (see [here](https://github.com/CitizenLabDotCo/citizenlab/blob/CL-339-i3-refactor-terminology-and-endpoint-usage/front/app/components/admin/PostManager/index.tsx#L438)). 

For the latter, the `Data` wrapper is somewhat easier, but the component itself is this massive >1000 line beast with more than 20 state variables and a bunch of manual subscriptions to streams. Would be great to refactor, but I believe this is something that we probably want to rewrite using @SaraBCL 's work on JSON forms, so maybe better to wait for that?

I have tried to keep the resource as minimal as possible here by reusing the logic and types from the complementary `useProjectAllowedInputTopics` hook. This way, it will be very easy to get rid of it in the future.

If you think it is valuable to still refactor all this stuff and get rid of the resource, I am up for it, but I will also have to ask my squad if it's ok wrt to planned release and the timeline of me trying to move to insights